props/saber_caiwyn_buttons.h

```diff
@@ -0,0 +1,604 @@
+//   Change Color: Rotate hilt while in Color Change Mode
+//   Color Zoom: Press & Hold Power and Rotate hilt while in Color Change Mode
+//   Save Changes and Exit Color Change Mode: Release Power
+//   Reset Color and Exit Color Change mode: Click Aux
+// Deactivate Saber: Hold Aux or Aux2 and press Power
+//
+// You will need the following sound files in order for menus to work properly:
+// vmbegin.wav - Enter Volume Change Menu
+// vmend01.wav - Save Volume Change
+// vmend02.wav - Cancel Volume Change
+// monce.wav - Set Track Player to play a single track one time
+// mloop.wav - Set Track Player to repeat a single track
+// mrotate.wav - Set Track Player to repeat all tracks
+// ccbegin.wav - Enter Color Change Mode
+// ccend01.wav - Save Color and Exit Color Change Mode
+// ccend02.wav - Reset Color and Exit Color Change Mode
+//
+// Options you can add to your config file:
+// #define DISABLE_BEEPS        - Disables the single beep that occurs when selecting fonts, tracks, or entering or exiting the volume and color change menus.
+//                                Note that beeps will still play if any sound file listed above is missing.
+// #define DISABLE_COLOR_CHANGE - Disables the color change menu.
+// #define CUSTOM_BATTERY_LEVEL - To use your own sound files when checking battery level instead of Proffie's built-in voice.
+//                                This will require additional sound files: mbatt.wav, mpercent.wav, mzero.wav, mnum1.wav - mnum20.wav, thirty.wav - ninety.wav, hundred.wav
+// #define DISABLE_BUTTON_CLASH - When you click the power button, both a clash sound and a lockup are simultaneously triggered.
+//                                This is because I have found that quickly starting and ending a lockup often sounds like multiple clashes rather than a single clash.
+//                                To smooth everything out, a clash sound is mixed in as well.  This works for most fonts, but if it sounds weird you can disable the clash sound and just use the lockup by itself.
+
+#ifndef PROPS_CAIWYN_BUTTONS_H
+#define PROPS_CAIWYN_BUTTONS_H
+
+#include "prop_base.h"
+#include "../sound/hybrid_font.h"
+#include "../sound/sound_library.h"
+
+#undef PROP_TYPE
+#define PROP_TYPE CaiwynButtons
+
+#ifndef NUM_BUTTONS
+#define NUM_BUTTONS 2
+#endif
+
+#if NUM_BUTTONS < 2
+#error /props/saber_caiwyn_buttons.h requires 2 or 3 Buttons for operation
+#endif
+
+#ifndef BUTTON_DOUBLE_CLICK_TIMEOUT
+#define BUTTON_DOUBLE_CLICK_TIMEOUT 500
+#endif
+
+#ifndef BUTTON_SHORT_CLICK_TIMEOUT
+#define BUTTON_SHORT_CLICK_TIMEOUT 500
+#endif
+
+#ifndef BUTTON_HELD_TIMEOUT
+#define BUTTON_HELD_TIMEOUT 500
+#endif
+
+#ifndef BUTTON_HELD_MEDIUM_TIMEOUT
+#define BUTTON_HELD_MEDIUM_TIMEOUT 1000
+#endif
+
+#ifndef BUTTON_HELD_LONG_TIMEOUT
+#define BUTTON_HELD_LONG_TIMEOUT 2000
+#endif
+
+EFFECT(vmbegin);
+EFFECT(vmend);
```

Comment:
vmbegin and vmend are available in the sound library, why not just use SayEnterVolumeMenu() / SayVolumeMenuEnd() ?

Reply:
The Say_____() functions invoke the wav_player buffer, which is used to string multiple sounds together.  The only time I use that is for the battery level check.  I used Effect() for these because I thought that was the more standard way of doing it, and because I do have monce.wav that isn't included in sound_library.  I can switch over to using the sound_library functions, though... I just have to figure out how to extend them with monce.wav

Reply:
Oh, now I remember.  I did it this way because it allows me to test for whether the sound files exist by doing "if (SFX_vmend)" and if the files don't exist I play a series of beeps instead.  I didn't see a way to do that in sound_library.  Is there one?

Reply:
Leaving this as is for now. I’m not opposed to using sound_library.Say***() to trigger sounds instead of hybrid_font.PlayPolyphonic(), but I would still need EFFECT() to create the SFX_*** variables that I test for the light the code, or some other alternative to test whether the wav files exist.

Reply:
I think the easy thing to do is just to tell people to install a voice pack, then you can basically assume that all of those files exists. No real need to test each file individually. Might be good to have an EFFECT for one of them, just to make sure that people didn't miss installing the voice pack though.

Reply:
Easy, yes, but I like having beep codes as a fallback in case sound files are missing or in the wrong format.  That's how the Color Wheel menu works already, and I modeled the behavior of my volume and track playback menus off that.  Is there some disadvantage to using EFFECT() to set up the sounds?  I notice this is how saber_sa22c_buttons.h sets up vmbegin and vmend, along with other sounds.

If there's another way to check to see if a file exists, I would be interesting in implementing it.

I did remove mbatt and mpercent from the list of EFFECT() definitions; I can just test for mnum, which is already defined in sound_library.

Reply:
EFFECTS take a little more memory, but almost never enough to matter.
If you want the fallbacks, I'm ok with that.